apps/client/src/pages/home/Home.page.tsx

```diff
@@ -36,30 +40,65 @@ export function Home(): ReactElement {
         onTokenReceived: () => navigate(`/${roomId}/room`),
       });
     } else {
-      // TODO: Create tournament in redux store
-      const tournamentId = getTournamentId();
+      // TODO: should receive tournament id(room id) from server
+      const roomId = getRoomId();
       initializeSocket({
         playerName,
         roomId,
-        onTokenReceived: () => navigate(`/${tournamentId}/room`),
+        onTokenReceived: () => navigate(`/${roomId}/room`),
       });
     }
   };
 
   const [playerName, setPlayerName] = useState<string>('');
+  const [isValidPlayerName, toggleIsValidPlayerName] = useState<boolean>(false);
+  // value of join room button. this value will be a room id
+  const [joinRoomButtonValue, setJoinRoomButtonValue] = useState<string>('');
+  const [isValidJoinRoomButtonValue, toggleIsValidRoomButtonValue] =
+    useState<boolean>(false);
   const [avtarURL, setAvtarURL] = useState<string>('');
 
   useEffect(() => {
+    if (playerNameSchema.safeParse(playerName).success) {
+      toggleIsValidPlayerName(true);
+    } else {
+      toggleIsValidPlayerName(false);
+      setAvtarURL('');
+      return;
+    }
+
     if (playerName === '') {
       setAvtarURL('');
     } else {
       setAvtarURL(generateAvatarLink(playerName));
     }
+
     return () => {
       setAvtarURL('');
     };
   }, [playerName]);
 
+  const roomIdChangeHandler = (value: string): void => {
+    setJoinRoomButtonValue(value);
+    // Room id is also tournament id without 'T:' prefix. Room id what players share.
+    const tournamentId = `T:${value}`;
+    if (tournamentIdSchema.safeParse(tournamentId).success) {
```

Comment:
Instead of using an if shall assign this to a variable and pass it directly to `toggleIsValidRoomButtonValue`

Reply:
Refactored 326958a300c64dc5f91b45f4ac5a6f62e2f2ecda